### PR TITLE
PSC-253

### PR DESCRIPTION
**Pull request by clewis1**
- Updating CompanyPscs to include relevant fields.
- Added Identification and NameElements entities.
- 
https://companieshouse.atlassian.net/browse/PSC-253

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImpl.java [hunk 1]
```diff
@@ -1,70 +1,91 @@
...
+
+    @Autowired
+    public CompanyPscsServiceImpl(RandomService randomService, CompanyPscsRepository repository, Clock clock) {
+        this.randomService = randomService;
+        this.repository = repository;
+        this.clock = clock;
+    }
 
     @Override
     public CompanyPscs create(CompanySpec spec) throws DataException {
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImpl.java:43]: Possible refactoring of this long method. Seems logical to introduce methods like `setLocation` for all the address-related lines, `setNaturesOfControl` for NOC lines, etc.

File: src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImplTest.java [hunk 2]
```diff
@@ -29,76 +42,116 @@ class CompanyPscsServiceImplTest {
...
         CompanyPscs returnedPsc = this.companyPscsService.create(spec);
 
         assertEquals(savedPsc, returnedPsc);
 
-        ArgumentCaptor<CompanyPscs> pscCaptor = ArgumentCaptor.forClass(CompanyPscs.class);
-        verify(repository).save(pscCaptor.capture());
+        verify(repository).save(argThat(this::hasExpectedValues));
+    }
+
+    private boolean hasExpectedValues(CompanyPscs companyPsc) {
```
Comment (hepsimo) [inline src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImplTest.java:80]: Possible refactor this long method with methods like `hasExpectedLocaton`, `hasExpectedNaturesOfControl` etc.

**Review by hepsimo — APPROVED** [review]
Changes look good. There are a couple of possible refactorings for long methods.

**Review by AlisonGriffiths — COMMENTED** [review]
Something to consider maybe for future work as we are limited on time now.
The code in TestDataServiceImpl.java which does the initial create(psc) feels like it could be improved.
It leads into the CompanyPSCSServiceImpl code and that differentiatePSC method which could be improved I think.
Perhaps something for discussion.
I am not holding up this merge, we can always look to refactoring that further at a later date.

**Pull request merged**